src/Haskakafka.hs

```diff
@@ -297,6 +307,15 @@ withKafkaConsumer configOverrides topicConfigOverrides brokerString tName partit
       destroyKafka kafka)
     (\(k, t) -> cb k t)
 
+withKafkaConf :: (KafkaConf -> IO a) -> IO a
+withKafkaConf =
```

Comment:
I've been out of this for a while, but don't you also need to modify the functions in  https://github.com/cosbynator/haskakafka/blob/f57568fa5cb1c8405be79ebbc631bc13b19c33f6/src/Haskakafka/InternalSetup.hs to function without a finalizer? Is ownership transferred?